src/domain/usecases/SendNotificationDataValuesUseCase.ts

```diff
@@ -0,0 +1,347 @@
+            period,
+            orgUnitName,
+            dataValues: dataValuesWithDeName,
+        };
+        const subjectTemplate = _.template(emailTemplate.subject);
+        const bodyTemplate = _.template(emailTemplate.body);
+        const subject = subjectTemplate(dataTemplate);
+        const body = bodyTemplate(dataTemplate);
+        return { body, subject };
+    }
+
+    private async saveExecutions(dataSetExecutions: NotificationStore, options: OptionsUseCase) {
+        await this.dataSetStoreRepository.save(dataSetExecutions, {
+            path: options.jsonExecutionsPath,
+            namespace: options.dataStoreNameSpace,
+        });
+    }
+
+    private getDataSetsExecutions(
+        dataSetKeys: string[],
+        settings: Settings,
+        dataSets: DataSet[],
+        orgUnits: OrgUnit[],
+        dataSetStore: NotificationStore | undefined
+    ) {
+        const keys = _(dataSetKeys)
+            .map(dataSetKey => {
+                const dataSetDetails = settings.dataSets[dataSetKey];
+                if (dataSetDetails) {
+                    return _(dataSetDetails.monitoring)
+                        .filter(m => m.enable)
+                        .map(monitor => {
+                            const dataSet = dataSets.find(
+                                dataSet =>
+                                    dataSet.id === dataSetKey ||
+                                    dataSet.name === dataSetKey ||
+                                    dataSet.code === dataSetKey
+                            );
+                            if (!dataSet) {
+                                throw Error(`Cannot find dataset: ${dataSetKey}`);
+                            }
+                            const orgUnit = orgUnits.find(
+                                ou =>
+                                    ou.id === monitor.orgUnit ||
+                                    ou.name === monitor.orgUnit ||
+                                    ou.code === monitor.orgUnit
+                            );
+
+                            if (!orgUnit) {
+                                throw Error(`Cannot find org unit: ${monitor.orgUnit}`);
+                            }
+                            return `${dataSet.id}-${orgUnit.id}-${monitor.period}`;
+                        })
+                        .value();
+                }
+                return undefined;
+            })
+            .compact()
+            .flatten()
+            .value();
+
+        const currentDate = new Date().toISOString();
+        const dataSetExecutions = keys.reduce<NotificationStore>((acum, key) => {
+            const storeInfo = dataSetStore ? dataSetStore[key] : undefined;
+            return {
+                ...acum,
+                [key]: {
+                    lastDateSentEmail: storeInfo?.lastDateSentEmail || "",
+                    lastUpdated: storeInfo?.lastUpdated || currentDate,
+                },
+            };
+        }, {});
```

Comment:
This pattern `reduce + set by key` is more declarative and type-safe using lodash `fromPairs`.